services/galley/src/Galley/Types/Clients.hs

```diff
@@ -24,68 +25,78 @@ import Data.Id
 import Data.Maybe (fromMaybe)
 import Data.Monoid
 import Data.Range
+import Galley.Types (UserClients (..))
 import Prelude hiding (filter)
 
 import qualified Data.Map.Strict as Map
 import qualified Data.Set        as Set
 
 newtype Clients = Clients
-    { clients :: Map UserId (Set ClientId)
+    { clients :: UserClients
     } deriving (Eq, Show, Monoid)
 
 instance Bounds Clients where
     within c x y =
-        let n = Map.size (clients c) in
+        let n = Map.size ((userClients . clients) c) in
         n >= fromIntegral x && n <= fromIntegral y
 
 null :: Clients -> Bool
-null = Map.null . clients
+null = Map.null . (userClients . clients)
 
 nil :: Clients
-nil = Clients Map.empty
+nil = Clients $ UserClients Map.empty
 
 userIds :: Clients -> [UserId]
-userIds = Map.keys . clients
+userIds = Map.keys . (userClients . clients)
 
 clientIds :: UserId -> Clients -> [ClientId]
-clientIds u c = Set.toList $ fromMaybe Set.empty (Map.lookup u (clients c))
+clientIds u c = Set.toList $ fromMaybe Set.empty (Map.lookup u ((userClients . clients) c))
 
 toList :: Clients -> [(UserId, [ClientId])]
-toList = Map.foldrWithKey' fn [] . clients
+toList = Map.foldrWithKey' fn [] . (userClients . clients)
   where
     fn u c a = (u, Set.toList c) : a
 
 fromList :: [(UserId, [ClientId])] -> Clients
-fromList = Clients . foldr fn Map.empty
+fromList = Clients . UserClients . foldr fn Map.empty
   where
     fn (u, c) = Map.insert u (Set.fromList c)
 
+fromUserClients :: UserClients -> Clients
+fromUserClients ucs = Clients ucs
+
```

Comment:
would `toClients` be more appropriate here since we are wrapping `UserClients` in `Clients`?

Reply:
I think from makes sense, since this is a way to construct Clients and it's in the Clients module, i.e. it makes sense to use this like `Clients.fromList ucs`.

Reply:
👍